src/map/attackround.cpp

```diff
@@ -199,12 +199,23 @@ void CAttackRound::DeleteAttackSwing()
 void CAttackRound::CreateAttacks(CItemWeapon* PWeapon, PHYSICAL_ATTACK_DIRECTION direction)
 {
     uint8 num = 1;
+    
+    bool isPC = m_attacker->objtype == TYPE_PC;
 
     // Checking the players weapon hit count
     if (PWeapon->getReqLvl() <= m_attacker->GetMLevel())
     {
         num = PWeapon->getHitCount();
     }
+    
+    // If the attacker is a mob, check to see if it has any special mutli-hit capabilties
+    if (!isPC)
```

Comment:
a pet could (no reason to not), but it certainly shouldn't be !isPC - you've no guarantee there won't be some other entity type that could end up in here that can't be cast to a mob entity